climada/util/plot.py

```diff
@@ -290,6 +293,79 @@ def geo_im_from_array(array_sub, coord, var_name, title,
 
     return axes
 
+def geo_scatter_categorical(array_sub, geo_coord, var_name, title,
+                            cat_name=None, pop_name=True, buffer=BUFFER,
+                            extend='neither', proj=ccrs.PlateCarree(),
+                            shapes=True, axes=None, **kwargs):
+    """Map plots for categorical data defined in array(s) over input
+    coordinates. The categories must be a finite set of unique
+    values as can be identified by np.unique() (int, float, strings, ...).
+    
+    The categories are shared among all subplots, i.e. are obtained from 
+    np.unique(array_sub). 
+    Example:
+        array_sub = [[1, 2, 1.0, 2], [1, 2, 'a', 'a']] 
+        -> categories mapping is [[0, 2, 1, 2], [0, 2, 3, 3]]
+    
+    
+    Parameters:
+        array_sub (np.array(1d or 2d) or list(np.array)): Each array (in a row
+            or in  the list) are categorical values at each point
+            in corresponding geo_coord that are ploted in one subplot.
+        coord (2d np.array): (lat, lon) for each point in a row.
+            The same grid is used for all subplots.
+        var_name (str or list(str)): label to be shown in the colorbar. If one
+            provided, the same is used for all subplots. Otherwise provide as
+            many as subplots in array_sub.
+        cat_name (list(str)), optional: List of category names in same order
+            as the ordered categories obtain from np.unique(array_sub). All
+            subplots are assumed to have the same categories. 
+            Default is numbers (1,2,...)
```

Comment:
How are you supposed to know the names if you don't know the categories that are determined automatically within this function? Furthermore, if `cat_name` is not supplied, it doesn't seem to be very useful to fall back to numbers. Instead, the unique values should be used as category names.

And if you know the categories before, you shouldn't use this function to re-define the categories. They might differ from your original categories. For example, if you think you have the categories 0-5, but you somehow truncated your data and, in fact, only the categories 3 and 4 appear in the data you provide to this function, your category names for 0-5 won't be matched correctly.

Reply:
Good points. One precision: it uses the unique valus as the category names, this is a bad Docstring. 

So, do you propose to not allow renaming the categories? One can always rename them afterwards from the axis handles (I would add this as example in the docstring since it is not obvious how too).

Reply:
I would rather have `cat_name` be a dict mapping from values to names.